### DISCUSSION TITLE
Present a different login form/theme depending on the RP that initiated the login request

### DISCUSSION THREAD

**Discussion by mig5** [ory/hydra#2921] [Q&A] [answered]
Hi,

Hydra currently seems to have a single config value/environment variable for the Login application.

My use case requires me to show a slightly different themed login form depending on which RP is in use. My RPs are effectively 'white-labelled' so that the login form matches the theme of the RP itself, but the OP/IDP is the same.

Is there any way I can get the Login app to infer which RP the login request began at - without relying on the HTTP Referer header (which won't be available due to the use of the Referrer-Policy security header, and which I'm not sure would work anyway, since I have a feeling the referrer would be the Hydra public API endpoint itself)?

https://github.com/ory/hydra/issues/1851 says "you can redirect in your login/consent app based on e.g. the client ID or some other parameter." It's not clear to me how to do that - can the `URLS_LOGIN` include some sort of interpolated query parameter that represents the client ID? As far as I can tell, all that the Login app 'sees' is the login_challenge query parameter, but no information as to what the RP was. Unless that's infer-able from the challenge itself?

**Comment by vinckr** [comment, marked as answer]
The login_challenge is used by the Login Endpoint to fetch information about the request.
If you look at the "Login Challenge Response" [here](https://www.ory.sh/hydra/docs/concepts/login/#the-login-endpoint), you can see that you get information on the client and some more. 

Does that help with your use case?

**Reply by mig5** [reply]
Thank you @vinckr, that is perfect. I didn't realise the Login Challenge response contained such a rich set of data about the oAuth2.0 client - including the redirectUris, client name, etc. Thanks again.
